lua/core/lualine/components.lua

```diff
@@ -0,0 +1,130 @@
+local conditions = require "core.lualine.conditions"
+local colors = require "core.lualine.colors"
+
+return {
+  vi_mode = {
+    function()
+      return " "
+    end,
+    left_padding = 0,
+    right_padding = 0,
+    condition = conditions.hide_in_width,
+  },
+  branch = {
+    "branch",
+    icon = " ",
+    condition = function()
+      return conditions.hide_in_width() and conditions.check_git_workspace()
```

Comment:
lualines `branch` component will check whether buffer is in a git workspace.
No need to check twice by placing it in `condition` .

Also since you guys have gitsigns you can just reuse the branch & diff info from gitsigns
instead of leaving lualine to figure them out .
https://github.com/shadmansaleh/lualine.nvim/wiki/Component-snippets#using-external-source-for-branch

Reply:
cc: @chaeing

Reply:
Thank you for this useful information :)